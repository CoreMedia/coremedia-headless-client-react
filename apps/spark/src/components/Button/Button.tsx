import React from "react";
import { useTranslation } from "react-i18next";
import styled from "styled-components";
import Link from "../Link/Link";

interface Props {
  linkTarget?: any;
  clickHandler?: () => void;
  text?: string;
  additionalClass?: string;
  openInNewTab?: boolean;
  externalLink?: boolean;
  primary?: boolean;
}

export const StyledButton = styled.div<{ primary?: boolean }>`
  background: ${(props) => (props.primary ? "var(--cta-primary-background)" : "var(--cta-background)")};
  border-radius: ${(props) => (props.primary ? "var(--cta-primary-border-radius)" : "var(--cta-border-radius)")};
  border: ${(props) => (props.primary ? "var(--cta-primary-border)" : "var(--cta-border)")};
  color: ${(props) => (props.primary ? "var(--cta-primary-text-color)" : "var(--cta-text-color)")};
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: var(--font-family-headline);
  font-size: var(--font-size-heading-3);
  padding: 10px 20px 5px;
  pointer-events: auto;
  position: relative;
  text-decoration: none;
  text-transform: uppercase;
  transition: all 0.5s ease;

  &:hover {
    background: ${(props) => (props.primary ? "var(--cta-primary-background-hover)" : "var(--cta-background-hover)")};
    border: ${(props) => (props.primary ? "var(--cta-primary-border-hover)" : "var(--cta-border-hover)")};
    color: ${(props) => (props.primary ? "var(--cta-primary-text-color-hover)" : "var(--cta-text-color-hover)")};
  }

  &:active,
  &:focus {
    outline: none;
    background: ${(props) => (props.primary ? "var(--cta-primary-background-active)" : "var(--cta-background-active)")};
    border: ${(props) => (props.primary ? "var(--cta-primary-border-active)" : "var(--cta-border-active)")};
    color: ${(props) => (props.primary ? "var(--cta-primary-text-color-active)" : "var(--cta-text-color-active)")};
    box-shadow:
      inset 0 0 0 1px var(--cta-background-active),
      inset 0 0 0 2px var(--cta-text-color-active);
  }

  & + & {
    margin-left: 5px;
  }
`;

const Button: React.FC<Props> = ({
  linkTarget,
  clickHandler,
  text,
  additionalClass,
  openInNewTab = false,
  externalLink = false,
  primary = true,
}) => {
  const { t } = useTranslation();

  if (!text) {
    // do not show default text "read more" for media / download links
    text =
      linkTarget && linkTarget.indexOf("/caas/v1/media") === 0 ? t("Button.downloadText") : t("Button.defaultText");
  }
  return (
    <StyledButton
      as={Link}
      to={linkTarget}
      clickHandler={clickHandler}
      className={additionalClass}
      role={"button"}
      openInNewTab={openInNewTab}
      externalLink={externalLink}
      primary={primary}
    >
      {text}
    </StyledButton>
  );
};

export default Button;
