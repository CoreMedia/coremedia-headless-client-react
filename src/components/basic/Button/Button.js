// @flow
import styled, { css } from 'styled-components';
import theme from 'styled-theming';

const buttonStyles = theme('name', {
  default: css`
    font-weight: 400;
    border: 2px solid;
    padding: 0.5em 1em;
    background: inherit;
  `,
  hkm: css`
    font-weight: 300;
    font-size: 0.8em;
    border: 1px solid #ea989d;
    padding: 5px 10px;
    color: #272727;
    background: linear-gradient(180deg, #fbe3e4, #fabfc0);
    text-transform: uppercase;
  `,
});

const Button = styled.button`
  ${buttonStyles};
  text-align: center;
  cursor: pointer;
`;
Button.displayName = 'Button';

Button.defaultProps = {
  theme: {
    name: 'default',
    breakpoints: {
      small: 480,
      tablet: 768,
      desktop: 992,
      large: 1280,
    },
  },
};

export default Button;
