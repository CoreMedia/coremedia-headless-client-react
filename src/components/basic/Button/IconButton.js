// @flow
import type { ComponentType } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

type Props = {
  color?: string,
  backgroundColor?: string,
};

const IconButton: ComponentType<Props> = styled.button`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 40px;
  height: 40px;
  padding: 7px;
  margin: 0;
  border-style: none;
  border-radius: 50%;
  border-width: 0;
  outline: none;
  color: ${props => props.color || 'inherit'};
  background-color: ${props => props.backgroundColor || 'transparent'};
  cursor: pointer;
`;

IconButton.displayName = 'IconButton';

IconButton.propTypes = {
  color: PropTypes.string,
  backgroundColor: PropTypes.string,
};

export default IconButton;
