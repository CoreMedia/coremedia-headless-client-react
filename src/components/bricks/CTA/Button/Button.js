// @flow
import styled from 'styled-components';

const Button = styled.button`
  border: 2px solid;
  margin-top: 1em;
  padding: 0.5em 1em;
  color: inherit;
  background-color: inherit;
`;
Button.displayName = 'Button';

export default Button;
