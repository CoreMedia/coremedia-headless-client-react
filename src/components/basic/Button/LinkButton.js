// @flow
import Button from './Button';

const LinkButton = Button.withComponent('a').extend`
  display: inline-block;
`;
LinkButton.displayName = 'LinkButton';

export default LinkButton;
