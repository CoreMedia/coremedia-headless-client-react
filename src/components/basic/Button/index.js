import Button from './Button';
import LinkButton from './LinkButton';
import IconButton from './IconButton';

export { Button, LinkButton, IconButton };
