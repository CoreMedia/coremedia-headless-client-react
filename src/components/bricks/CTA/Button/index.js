import Button from './Button';
import LinkButton from './LinkButton';

export { Button, LinkButton };
