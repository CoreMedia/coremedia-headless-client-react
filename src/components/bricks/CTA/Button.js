// @flow
import { Button as BasicButton, LinkButton as BasicLinkButton } from '../../basic/Button/';

const Button = BasicButton.extend`
  margin-top: 1em;
`;

const LinkButton = BasicLinkButton.extend`
  margin-top: 1em;
`;

export { Button, LinkButton };
