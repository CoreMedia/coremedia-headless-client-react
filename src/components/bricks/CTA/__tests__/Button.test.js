// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Button } from '../Button';

describe('Button Component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Button>Content</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
