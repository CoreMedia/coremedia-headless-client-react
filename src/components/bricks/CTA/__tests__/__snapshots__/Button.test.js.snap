// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly 1`] = `
.c0 {
  border: 2px solid;
  margin-top: 1em;
  padding: 0.5em 1em;
  color: inherit;
  background-color: inherit;
}

<button
  className="c0"
>
  Content
</button>
`;
